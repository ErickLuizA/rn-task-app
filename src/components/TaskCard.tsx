import React from 'react'
import { StyleSheet, View, Text } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import { useTheme } from 'react-native-paper'
import { MaterialIcons } from '@expo/vector-icons'

import Task from '../models/Task'

interface ITaskCardProps {
  taskType: string
  data: Task
}

export default function TaskCard({ taskType, data }: ITaskCardProps) {
  const { colors } = useTheme()

  return (
    <TouchableOpacity>
      {data && (
        <View style={[styles.card, { backgroundColor: colors.primary }]}>
          <Text style={[{ color: colors.text }, styles.title]}>
            {data?.name}
          </Text>
          <Text style={[{ color: colors.secondary }, styles.category]}>
            {data?.category}
          </Text>
          <View style={styles.row}>
            {taskType === 'Daily' ? (
              <Text style={[{ color: colors.secondary }, styles.hour]}>
                {data.date.toLocaleTimeString().slice(0, 5)}
              </Text>
            ) : (
              <Text style={[{ color: colors.secondary }, styles.date]}>
                {data.date.toLocaleDateString()}{' '}
              </Text>
            )}
            <MaterialIcons
              name={
                data.date.toLocaleTimeString() > '18'
                  ? 'brightness-3'
                  : 'brightness-7'
              }
              color={colors.secondary}
              size={40}
            />
          </View>
        </View>
      )}
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  card: {
    width: '100%',
    padding: 10,
    borderRadius: 4,
  },

  title: {
    fontSize: 20,
    fontFamily: 'Roboto-BoldItalic',
  },

  category: {
    fontSize: 18,
    fontFamily: 'Roboto-Regular',
  },

  hour: {
    fontSize: 18,
    fontFamily: 'Roboto-LightItalic',
  },

  date: {
    fontSize: 18,
    fontFamily: 'Roboto-LightItalic',
  },

  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
})
