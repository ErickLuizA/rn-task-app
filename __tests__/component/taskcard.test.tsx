import * as React from 'react'
import { render } from 'react-native-testing-library'
import TaskCard from '../../src/components/TaskCard'

describe('<TaskCard />', () => {
  it('render the TaskCard component', () => {
    const tree = render(<TaskCard taskType="Daily" />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
