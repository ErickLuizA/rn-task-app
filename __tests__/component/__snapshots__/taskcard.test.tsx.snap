// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskCard /> render the TaskCard component 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
  testID="cardContainer"
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#f50057",
        "borderRadius": 15,
        "color": "#ffffff",
        "fontFamily": "System",
        "fontSize": 14,
        "fontWeight": "400",
        "height": 30,
        "lineHeight": 30,
        "minWidth": 30,
        "opacity": 0,
        "overflow": "hidden",
        "paddingHorizontal": 20,
        "textAlign": "center",
        "textAlignVertical": "center",
        "transform": Array [
          Object {
            "rotate": "270deg",
          },
        ],
      }
    }
  >
    21/07
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 10,
          "paddingVertical": 10,
          "width": 576.9230769230769,
        },
        Object {
          "backgroundColor": "#6200ee",
        },
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
            },
          ]
        }
      >
         
        Task Name
         
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": undefined,
            },
            Object {
              "fontFamily": "Roboto-Light",
              "fontSize": 18,
            },
          ]
        }
      >
         
        Category Name
         
      </Text>
    </View>
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 40,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        îŽ¨
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": undefined,
            },
            Object {
              "fontFamily": "Roboto-LightItalic",
              "fontSize": 18,
            },
          ]
        }
      >
         12:30
      </Text>
    </View>
  </View>
</View>
`;
